src/structures/GuildChannel.js

```diff
@@ -95,13 +96,33 @@ class GuildChannel extends Channel {
    */
   get permissionsLocked() {
     if (!this.parent) return null;
-    if (this.permissionOverwrites.size !== this.parent.permissionOverwrites.size) return false;
-    return this.permissionOverwrites.every((value, key) => {
-      const testVal = this.parent.permissionOverwrites.get(key);
+
+    // Get all overwrites
+    const overwriteIds = new Set([...this.permissionOverwrites.keys(), ...this.parent.permissionOverwrites.keys()]);
+
+    // Compare all overwrites
+    return [...overwriteIds].every(key => {
+      const channelVal = this.permissionOverwrites.get(key);
+      const parentVal = this.parent.permissionOverwrites.get(key);
+
+      // Handle empty overwrite
+      if (
+        (channelVal === undefined &&
+          parentVal.deny.bitfield === BitField.defaultBit &&
+          parentVal.allow.bitfield === BitField.defaultBit) ||
+        (parentVal === undefined &&
+          channelVal.deny.bitfield === BitField.defaultBit &&
+          channelVal.allow.bitfield === BitField.defaultBit)
```

Comment:
BitField.defaultBit is 0
Permissions.defaultBit is 0n

This will never match with `===`, although APPARENTLY will with `==` (thanks JS.)